plugin/main/src/kotlinx/benchmark/gradle/internal/BenchmarkDependencies.kt

```diff
@@ -0,0 +1,79 @@
+package kotlinx.benchmark.gradle.internal
+
+import kotlinx.benchmark.gradle.BenchmarksExtension
+import org.gradle.api.*
+import org.gradle.api.artifacts.*
+import org.gradle.api.attributes.*
+import org.gradle.api.attributes.Usage.*
+import org.gradle.api.model.*
+import org.gradle.util.GradleVersion
+
+/**
+ * Utility for containing all Gradle [Configuration]s used by Benchmarks.
+ */
+internal class BenchmarkDependencies(
+    project: Project,
+    benchmarksExtension: BenchmarksExtension,
+) {
+    private val objects: ObjectFactory = project.objects
+
+    private val benchmarkGenerator: Configuration =
+        project.configurations.create("benchmarkGenerator") {
+            it.description =
+                "Internal Kotlinx Benchmarks Configuration. Contains declared dependencies required for running benchmark generators."
```

Comment:
Maybe `kotlinx-benchmark` instead? I don't think we refer to the plugin as `Kotlinx Benchmarks` anywhere else.